internal/translation/privateendpoint/privateendpoint.go

```diff
@@ -0,0 +1,181 @@
+/*
+Copyright 2024 MongoDB.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package privateendpoint
+
+import (
+	"context"
+	"strings"
+
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/cmp"
+)
+
+const (
+	ErrorServiceNotFound = "PRIVATE_ENDPOINT_SERVICE_NOT_FOUND"
+)
+
+type PrivateEndpointService interface {
+	ListPrivateEndpoints(ctx context.Context, projectID, provider string) ([]EndpointService, error)
+	GetPrivateEndpoint(ctx context.Context, projectID, provider, ID string) (EndpointService, error)
+	CreatePrivateEndpointService(ctx context.Context, projectID string, peService EndpointService) (EndpointService, error)
+	DeleteEndpointService(ctx context.Context, projectID, provider, ID string) error
+	CreatePrivateEndpointInterface(ctx context.Context, projectID, provider, serviceID, gcpProjectID string, peInterface EndpointInterface) (EndpointInterface, error)
+	DeleteEndpointInterface(ctx context.Context, projectID, provider, serviceID, ID string) error
+}
+
+type PrivateEndpoint struct {
+	api admin.PrivateEndpointServicesApi
+}
+
+func (pe *PrivateEndpoint) ListPrivateEndpoints(ctx context.Context, projectID, provider string) ([]EndpointService, error) {
+	services, _, err := pe.api.ListPrivateEndpointServices(ctx, projectID, provider).
+		Execute()
+	if err != nil {
+		return nil, err
+	}
+
+	peServices := make([]EndpointService, 0, len(services))
+	for _, service := range services {
+		interfaceIDs := getInterfacesIDs(&service)
+		peInterfaces := make([]EndpointInterface, 0, len(interfaceIDs))
+
+		for _, interfaceID := range interfaceIDs {
+			peInterface, err := pe.getEndpointInterfaces(ctx, projectID, provider, service.GetId(), interfaceID)
+			if err != nil {
+				return nil, err
+			}
+
+			if peInterface != nil {
+				peInterfaces = append(peInterfaces, peInterface)
+			}
+		}
+
+		cmp.NormalizeSlice(peInterfaces, func(a, b EndpointInterface) int {
+			return strings.Compare(a.InterfaceID(), b.InterfaceID())
+		})
```

Comment:
this effectively is just sorting, not a complex nested struct normalization, right? If so, why is simple sorting not good enough (and more explicit) here?